clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,11 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            if (!this.subscriptions.allConsumed().isEmpty() && this.autoCommitEnabled)
```

Comment:
I was thinking if there is no records consumed by the consumer (no `poll`) before calling `unsubscribe()` there is no need for a commit. I think we could change the check in `assign` method to this too. What do you think?